QN 4   TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


In [5]:
import pandas as pd
import numpy as np
import scipy.stats as stats

In [6]:
df4=pd.read_csv('Costomer+OrderForm (1).csv')
df4

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [7]:
#ways you can get 

In [8]:
phi=df4.iloc[:,0]
phi

0      Error Free
1      Error Free
2      Error Free
3      Error Free
4      Error Free
          ...    
295    Error Free
296    Error Free
297    Error Free
298    Error Free
299    Error Free
Name: Phillippines, Length: 300, dtype: object

In [9]:
df4.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [10]:
df4.groupby('Phillippines').size()

Phillippines
Defective      29
Error Free    271
dtype: int64

In [11]:
phi=df4.iloc[:,0].value_counts()
phi

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [12]:
print(df4.iloc[:,0].value_counts(),df4.iloc[:,1].value_counts(),df4.iloc[:,2].value_counts(),df4.iloc[:,3].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 Error Free    269
Defective      31
Name: Malta, dtype: int64 Error Free    280
Defective      20
Name: India, dtype: int64


In [13]:
arr=[df4.iloc[:,0].value_counts(),df4.iloc[:,1].value_counts(),df4.iloc[:,2].value_counts(),df4.iloc[:,3].value_counts()]

In [14]:
type(arr)

list

In [15]:
arr

[Error Free    271
 Defective      29
 Name: Phillippines, dtype: int64, Error Free    267
 Defective      33
 Name: Indonesia, dtype: int64, Error Free    269
 Defective      31
 Name: Malta, dtype: int64, Error Free    280
 Defective      20
 Name: India, dtype: int64]

In [16]:
data_array=np.array([series.values for series in arr])
data_array

array([[271,  29],
       [267,  33],
       [269,  31],
       [280,  20]])

In [17]:
print(data_array)

[[271  29]
 [267  33]
 [269  31]
 [280  20]]


In [18]:
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [19]:
chi=stats.chi2_contingency(obs)
chi

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

In [20]:
p_value=chi[1]   #p value
p_value

0.2771020991233135

**conl:** As comparince to p value and α value = 0.05 p value > α it's 0.27710 > 0.05

Thus,Accept H0, hence customer order forms defective % does not varies by centre.